src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -313,3 +315,63 @@ def get_src_py_file_to_dest_py_file_map(
         # TODO: Does this case 5 need to be considered?
 
         return src_py_file_to_dest_py_file_map
+
+
+def _create_new_sql_file_at_path(parent_dir: Path, file_name: str) -> Optional[Path]:
+    """
+    Creates an empty SQL file at the same location specified as the parent directory, with file_name as part of the new file's name.
+    """
+    sql_file_name = f"__{file_name}_{uuid.uuid4().hex}.sql"
+    sql_file_path = Path(parent_dir, sql_file_name)
+    try:
+        f = open(sql_file_path, "x")
+        f.close()
+    except FileExistsError:
+        # It is unlikely that a user has created a file with this name, given the use of UUID.
+        cc.warning(
+            f"File {sql_file_path} already exists, cannot overwrite existing files."
+        )
+        cc.warning("Skipping creation of this file.")
+        return None
+    return sql_file_path
+
+
+def create_and_write_to_sql_files(dest_file_py_file_to_ddl_map: Dict[Path, str]):
+    """
+    Creates SQL files at predetermined location as specified in the input dictionary,
+    and writes the contents from the map to the new file.
+    """
+    dest_file_to_sql_file: Dict[Path, Path] = {}
+    for dest_file, all_ddl in dest_file_py_file_to_ddl_map.items():
+        if all_ddl is None or all_ddl.strip() == "":
+            continue
+
+        sql_file_path = _create_new_sql_file_at_path(
+            parent_dir=dest_file.parent, file_name=dest_file.name.split(".")[0]
+        )
+        if sql_file_path is None:
+            # We lose information in this step, since the file already exists and we do not overwrite or append to that
+            # as we do not know what already exists in that file.
+            continue
+
+        with open(sql_file_path, "a") as file:
```

Comment:
"a" looks odd here, if we're generating we'll want to overwrite the file each time when we use a stable naming scheme.